### PR TITLE
Story of components

### LINKED ISSUE #104: Fix Storybook test config
When you run `pnpm test:storybook`, warnings and an error are thrown for issues with the config & vitest.

```
> first-spark-app@0.0.1 test:storybook /Users/jonah/src/fs/first-spark-app
> vitest --project storybook

(node:9438) ExperimentalWarning: Type Stripping is an experimental feature and might change at any time
(Use `node --trace-warnings ...` to show where the warning was created)
WARN No story files found for the specified pattern: src/**/*.mdx
Warning: Starting in Storybook 8.5.0-alpha.18, the "test.include" option in Vitest is discouraged in favor of just using the "stories" field in your Storybook configuration.

The values you passed to "test.include" will be ignored, please remove them from your Vitest configuration where the Storybook plugin is applied.

More info: https://github.com/storybookjs/storybook/blob/next/MIGRATION.md#indexing-behavior-of-storybookexperimental-addon-test-is-changed

 DEV  v3.1.2 /Users/jonah/src/fs/first-spark-app

Loaded  vitest@3.1.2  and  @vitest/browser@3.0.8 .
Running mixed versions is not supported and may lead into bugs
Update your dependencies and make sure the versions match.
The following Vite config options will be overridden by SvelteKit:
  - base
Warning: Starting in Storybook 8.5.0-alpha.18, the "test.include" option in Vitest is discouraged in favor of just using the "stories" field in your Storybook configuration.

The values you passed to "test.include" will be ignored, please remove them from your Vitest configuration where the Storybook plugin is applied.

More info: https://github.com/storybookjs/storybook/blob/next/MIGRATION.md#indexing-behavior-of-storybookexperimental-addon-test-is-changed
⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯ Unhandled Errors ⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯

Vitest caught 1 unhandled error during the test run.
This might cause false positive tests. Resolve unhandled errors to make sure your tests are not affected.

⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯ Unhandled Error ⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯
TypeError: project.vitest._browserSessions.createAsyncSession is not a function
```

### LINKED ISSUE #105: Complete AuthCard stories
What value does this example provide? The props are not mocked, you can't actually do anything or visualize any children.
> Image

Perhaps [use the base component that this AuthCard thing wraps](https://next.shadcn-svelte.com/docs/components/card) as a potential example?

### LINKED ISSUE #106: Complete the LightSwtich Stories
- It should be called LightSwitch, and is currently ThemeButton
- Why doesn't it work?

### LINKED ISSUE #107: Complete the NavBar stories
What value does this example provide?
> Image

- Why aren't you able to customize the size?
- Why doesn't auth work?
- Why can't we just toggle where you're authenticated or not and have one complete story?
- Are there tests for this component?

### LINKED ISSUE #109: Complete the LanguageButton stories
Add some sort of text that can be translated, something that will also showcase a LTR/RTL orientation switch

### PR DESCRIPTION
**Pull request by RaghvindYadav**
## What does this PR do?
This branch contains story of components and also some tests. 

### Related Issues & PRs

The following issues are fixed in this PR:

- Closes #106 
- Closes #104 
- Closes #105 
- Closes #107 
- Closes #109 

### Limitations & Concerns

- **Known limitations**:
- Language switch test running in infinite loop. Change theme does't change the theme in actual.
- **Edge cases**:
- **Potential risks**:

### How to Test

1. Prerequisites:
 Install the dependencies.
2. Test steps:
   run the following terminal code `pnpm test: storybook`
3. Expected results:
   All component test will pass, except pages.
Screenshot 2025-05-15 at 6 23 24 PM

## Final Checklist
- [x] Linked related issues
- [x] Rebased with target branch
- [x] No lint errors
- [ ] QA'd in multiple browsers
- [x] Added/updated tests (if applicable)
- [ ] Mobile responsive (if applicable)
- [ ] Updated documentation (if applicable)
- [x] Notified the Team of this updates readiness

---

## AI Usage

- [ ] AI tools were used in development
  - **Tool used**:
    Copilot is used to write the similar code of other tests.
  - **Prompts used**:
    Please write the story and test of the <tagged component>. Here is svelte5 llm instruction, please follow it.
    - **Initial prompt**: 
- [ ] No AI tools were used
  - **Reason**:
    

## References

- **Documentation**:
- **Discussions**:
- **Related PRs**: #86

### CODE REVIEW

File: src/stories/components/ConnectionSonner.stories.svelte [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+  });
+</script>
+
+<Story
+  name="Offline"
+  args={{ clientConnection: true }}
+  play={async ({ canvasElement }) => {
+    // Should show offline toast
+    const canvas = within(canvasElement);
+    expect(canvas.getByText(/offline/i)).toBeInTheDocument();
```
[the remark this answers is not in the document]
Comment (RaghvindYadav) [reply src/stories/components/ConnectionSonner.stories.svelte]: nice catch what do you think @jonahtwalker?

File: public/favicon.png
Comment (RaghvindYadav) [inline public/favicon.png:1]: @jonahtwalker I need a confirmation to keep the file in public directory. When I wrote the story for app sidebar I was getting the not found error for the images and thats why I did it.

File: .gitignore
Comment (RaghvindYadav) [inline .gitignore:1]: this changes was added automatically by playwright configuration.

**Comment by jonahtwalker** [comment]
@RaghvindYadav I have rebased this.

We mentioned updates based on the language earlier today and it seems like those are most of the issues in the story test.

On another note the controls are looking good! That's what we've had in mind to be able to doctor up an odd test case to share easily.

Once the stories are repaired this should be good to go. 👍🏼

**Comment by jonahtwalker** [comment]
Is this supposed to replace your #86? It's listed above as `Closes`

Further, some of these can be combined. Like, we don't need two NavBar stories - we should have a control to become authenticated. 

The language button seems to work, why doesn't the light switch?

If we aren't going to complete those stories let's remove them from the description.

File: src/stories/pages/SignInPage.stories.svelte [hunk 2]
```diff
@@ -6,71 +6,86 @@
...
+    try {
+      await userEvent.click(signInButton);
+    } catch (error) {
+      console.log('Button click was blocked, using programmatic submit instead');
+      // Fallback: trigger form submission programmatically
+      const form = canvasElement.querySelector('form');
+      if (form) {
+        form.dispatchEvent(new Event('submit', { bubbles: true, cancelable: true }));
+      }
+    }
```
[the remark this answers is not in the document]
Comment (RaghvindYadav) [reply src/stories/pages/SignInPage.stories.svelte]: it's related to page story, I will skip that.

**Comment by RaghvindYadav** [comment]
> Is this supposed to replace your #86? It's listed above as `Closes`

No replace of #86 is #130.

> we don't need two NavBar stories - we should have a control to become authenticated.

It's not two stories, it a two case of test when not signedIn and when signedIn.

**Comment by jonahtwalker** [comment]
This is incomplete but largely in place, I am merging it in anyways.
